server/protocols/webadmin/webadmin-mailbox-deleted-message-vault/src/main/java/org/apache/james/webadmin/vault/routes/DeletedMessagesVaultRestoreTask.java

```diff
@@ -72,24 +88,32 @@ void incrementErrorRestoreCount() {
     private final User userToRestore;
     private final RestoreService vaultRestore;
     private final AdditionalInformation additionalInformation;
+    private final String requestBody;
+    private final QueryDTO queryDTO;
 
-    DeletedMessagesVaultRestoreTask(User userToRestore, RestoreService vaultRestore) {
+    DeletedMessagesVaultRestoreTask(String requestedQuery, QueryDTO queryDTO, User userToRestore, RestoreService vaultRestore) {
+        this.requestBody = requestedQuery;
         this.userToRestore = userToRestore;
         this.vaultRestore = vaultRestore;
-        this.additionalInformation = new AdditionalInformation(userToRestore);
+        this.additionalInformation = new AdditionalInformation(userToRestore, requestedQuery);
+        this.queryDTO = queryDTO;
     }
 
     @Override
     public Result run() {
         try {
-            return vaultRestore.restore(userToRestore).toStream()
+            return vaultRestore.restore(userToRestore, queryDTO).toStream()
                 .peek(this::updateInformation)
                 .map(this::restoreResultToTaskResult)
                 .reduce(Task::combine)
                 .orElse(Result.COMPLETED);
         } catch (MailboxException e) {
             LOGGER.error("Error happens while restoring user {}", userToRestore.asString(), e);
             return Result.PARTIAL;
+        } catch (QueryParser.QueryParserException e) {
```

Comment:
IMO the query parsing is a reponsibility of the route. Parsing should already be done when creating the task.